File: go/lasagna-master/lasagna_master.go
package lasagna

// PreparationTime estimates how long the preparation will take.
func PreparationTime(layers []string, avgTimePerLayer int) int {
	if avgTimePerLayer == 0 {
		avgTimePerLayer = 2
	}
	return len(layers) * avgTimePerLayer
}

// Quantities determines the quantity of noodles and sauce needed.
func Quantities(layers []string) (noodles int, sauce float64) {
	for _, layer := range layers {
		if layer == "noodles" {
			noodles += 50
		} else if layer == "sauce" {
			sauce += 0.2
		}
	}
	return
}

// AddSecretIngredient accepts two string slices as parameters
// replaces the last element in the second slice with the last
// element from the first slice.
func AddSecretIngredient(friendsList, myList []string) {
	secretIngredient := friendsList[len(friendsList)-1]
	myList[len(myList)-1] = secretIngredient
}

// ScaleRecipe calculates the amounts for different numbers of portions.
func ScaleRecipe(quantitiesForTwo []float64, numberOfPortions int) []float64 {
	quantities := make([]float64, len(quantitiesForTwo))
	for i, quantity := range quantitiesForTwo {
		quantities[i] = quantity / 2 * float64(numberOfPortions)
	}
	return quantities
}

File: javascript/lasagna-master/lasagna-master.js
export function cookingStatus(remainingTime) {
    if (remainingTime === undefined)
        return 'You forgot to set the timer.'
    if (remainingTime === 0)
        return 'Lasagna is done.'
    return 'Not done, please wait.'
}

export function preparationTime(layersArray, minutesPerLayer=2) {
    let number = Number(minutesPerLayer)
    return layersArray.length * number
}

export function quantities(layersArray) {
    let amounts = { noodles: 0, sauce: 0 }
    for (let layer of layersArray) {
        if (layer == 'noodles') {
            amounts.noodles += 50;
        } else if (layer == 'sauce') {
            amounts.sauce += 0.2;
        }
    }
    return amounts
}

export function addSecretIngredient(friendsList, myList) {
    myList.push(friendsList[friendsList.length - 1])
}

export function scaleRecipe(recipe, portions) {
    let newRecipe = {}
    for (let ingredient in recipe) {
        newRecipe[ingredient] = recipe[ingredient] / 2 * portions
    }
    return newRecipe
}